src/Hl7.Fhir.Shims.Base/Specification/Terminology/CodeSystemFilterProcessor.cs

```diff
@@ -54,32 +56,56 @@ private static List<CodeSystem.ConceptDefinitionComponent> applyFilters(List<Val
             return result;
         }
 
-        private static IEnumerable<CodeSystem.ConceptDefinitionComponent> applyFilter(List<CodeSystem.ConceptDefinitionComponent> concepts, List<CodeSystem.PropertyComponent> properties, ValueSet.FilterComponent filter)
+        private static IEnumerable<CSDC> applyFilter(List<CSDC> concepts, List<CodeSystem.PropertyComponent> properties, ValueSet.FilterComponent filter)
         {
             return filter.Op switch
             {
                 FilterOperator.IsA => applyIsAFilter(concepts, properties, filter),
-                _ => throw new ValueSetExpansionTooComplexException($"ConceptSets with a filter {filter.Op} are not yet supported.")
+                FilterOperator.IsNotA => applyIsNotAFilter(concepts, properties, filter),
+                FilterOperator.DescendentOf => applyDescentantOfFilter(concepts, properties, filter),
+                _ => throw new ValueSetExpansionTooComplexException($"ConceptSets with filter `{filter.Op.GetLiteral()}` are not yet supported.")
             };
         }
 
-        private static IEnumerable<CodeSystem.ConceptDefinitionComponent> applyIsAFilter(List<CodeSystem.ConceptDefinitionComponent> concepts, List<CodeSystem.PropertyComponent> properties, ValueSet.FilterComponent filter)
+        private static IEnumerable<CSDC> applyDescentantOfFilter(List<CSDC> concepts, List<CodeSystem.PropertyComponent> properties, ValueSet.FilterComponent filter)
```

Comment:
Looks like these functions have the same "body", but can be parameterized using a `Func<>` to make it one function. Not sure, but looks like it.